patches/tModLoader/Terraria/ModLoader/Config/EntityDefinition.cs

```diff
@@ -12,6 +12,7 @@ namespace Terraria.ModLoader.Config;
 
 /// <summary>
 /// Classes implementing EntityDefinition serve to function as a way to save and load the identities of various Terraria objects. Only the identity is preserved, no other data such as stack size, damage, etc. These classes are well suited for ModConfig, but can be saved and loaded in a TagCompound as well.
+/// <para/> An EntityDefinition can potentially refer to content removed from a mod or content from a mod not currently enabled. Use <see cref="IsUnloaded"/> to check if this is the case, especially before using <see cref="Type"/> as an index into an array. It is usually desirable to preserve all unloaded EntityDefinition entries so user data is not lost, so structure your code accordingly.
```

Comment:
I assume these doc updates were from the config example. I won't make you remove it, just doesn't really belong in this PR